lib/chef/property.rb

```diff
@@ -110,6 +110,9 @@ def initialize(**options)
         raise ArgumentError, "Cannot specify both default and name_property/name_attribute together on property #{self}"
       end
 
+      # Freeze the default if it isn't a lazy value.
+      default.freeze unless default.is_a?(DelayedEvaluator)
```

Comment:
what about hashes and arrays?  pretty sure the yum_package default mutation bug was on a hash key.